tests/signals/tests.py

```diff
@@ -351,3 +351,24 @@ def test_register_model_class_senders_immediately(self):
         apps2 = Apps()
         signals.post_init.connect(self.receiver, sender=Book, apps=apps2)
         self.assertEqual(list(apps2._pending_operations), [])
+
+
+class ModelStateTests(TestCase):
+    databases = ['default', 'other']
+
+    def setUp(self):
+        a1 = Author.objects.using('other').create(id=1)
+        b1 = Book.objects.using('other').create(id=1)
+        b1.authors.set([a1])
+        signals.post_init.connect(self.post_signal, sender=Book)
+
+    def tearDown(self):
+        signals.post_init.disconnect(self.post_signal, sender=Book)
+
+    def post_signal(self, instance, **kwargs):
+        self.assertFalse(instance._state.adding)
+        self.assertEqual(instance._state.db, 'other')
+        self.assertEqual(instance._state.db, instance.authors.get()._state.db)
+
+    def test_model_state(self):
+        Book.objects.using('other').get(pk=1)
```

Comment:
Hello @nasirhjafri. Wouldn't it be better to end with assertion this case? 

sorry about my comment if it is unnecessary. I'm trying to learn the mechanism.

Reply:
Hi @cansarigol, Doing a get query on `Book` will trigger the signal to `post_signal`. The issue was in the signals, so assertions are there. Kindly see the `post_signal` function at line 368.

Reply:
Thanks again for answer. I had thought like this.

```
def test_model_state(self):
        def post_signal(instance, **kwargs):
            import copy
            self.instance = copy.deepcopy(instance)

        signals.post_init.connect(post_signal, sender=Book)
        Book.objects.using('other').get(pk=1)
        self.assertFalse(self.instance._state.adding)
        self.assertEqual(self.instance._state.db, 'other')
        self.assertEqual(self.instance._state.db, self.instance.authors.get()._state.db)
        signals.post_init.disconnect(post_signal, sender=Book)
```

Reply:
Thank, I've updated the tests.